src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/CompatConfig.java

```diff
@@ -100,4 +100,28 @@ public boolean transportInterClusterAuthEnabled() {
             return true;
         }
     }
+
+    /**
+     * Returns true if passive transport auth is enabled
+     */
+    public boolean transportInterClusterPassiveAuthEnabled() {
+        final boolean interClusterAuthInitiallyPassive = staticSettings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_PASSIVE_INTERTRANSPORT_AUTH_INITIALLY, false);
+
+            if(interClusterAuthInitiallyPassive) {
+                if(dcm == null) {
```

Comment:
`dcm` is null when the security index has not been initialised.
We are returning false as it is the default value, and the reason we are returning the default value here is because other feature flags are also doing the same and I wanted the behaviour to be consistent.
Adding to that, with the current flow, the only side effect it causes when it is null is that the exception thrown is not very clear of what is actually happening i.e. it will throw a "No user found for" instead of something like "Security index not initialised"

If we want to instead throw a runtime exception here, then the concerns I have are:
1. What about the other places where we return a default value? Do we make changes there too to have a consistent experience?
2. What if a caller wants to handle it differently instead of throwing an exception. I can't think of a case yet but want to have that discussion. On similar lines, what if the caller wants to throw a different exception or the corresponding exception message? For both these cases, if we agree caller needs to have more freedom on how to treat the null value, perhaps we can return a NULL value instead so caller can decide.

@vrozov - Would like your input here.

Reply:
I am OK with the current approach as it is consistent with the prior behavior. If it needs to be changed, we can do it in a separate PR for all cases.